invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/ModelManagerPanel/ModelList.tsx

```diff
@@ -31,47 +31,89 @@ const ModelList = (props: ModelListProps) => {
-    useState<CombinedModelFormat>('images');
-
-  const { filteredDiffusersModels } = useGetMainModelsQuery(ALL_BASE_MODELS, {
-    selectFromResult: ({ data }) => ({
-      filteredDiffusersModels: modelsFilter(
-        data,
-        'main',
-        'diffusers',
-        nameFilter
-      ),
-    }),
-  });
-
-  const { filteredCheckpointModels } = useGetMainModelsQuery(ALL_BASE_MODELS, {
-    selectFromResult: ({ data }) => ({
-      filteredCheckpointModels: modelsFilter(
-        data,
-        'main',
-        'checkpoint',
-        nameFilter
-      ),
-    }),
-  });
-
-  const { filteredLoraModels } = useGetLoRAModelsQuery(undefined, {
-    selectFromResult: ({ data }) => ({
-      filteredLoraModels: modelsFilter(data, 'lora', undefined, nameFilter),
-    }),
-  });
+    useState<CombinedModelFormat>('all');
+
+  const { filteredDiffusersModels, isLoadingDiffusersModels } =
+    useGetMainModelsQuery(ALL_BASE_MODELS, {
+      selectFromResult: ({ data, isLoading }) => ({
+        filteredDiffusersModels: modelsFilter(
+          data,
+          'main',
+          'diffusers',
+          nameFilter
+        ),
+        isLoadingDiffusersModels: isLoading,
+      }),
+    });
+
+  const { filteredCheckpointModels, isLoadingCheckpointModels } =
+    useGetMainModelsQuery(ALL_BASE_MODELS, {
+      selectFromResult: ({ data, isLoading }) => ({
+        filteredCheckpointModels: modelsFilter(
+          data,
+          'main',
+          'checkpoint',
+          nameFilter
+        ),
+        isLoadingCheckpointModels: isLoading,
+      }),
+    });
+
+  const { filteredLoraModels, isLoadingLoraModels } = useGetLoRAModelsQuery(
+    undefined,
+    {
+      selectFromResult: ({ data, isLoading }) => ({
+        filteredLoraModels: modelsFilter(data, 'lora', undefined, nameFilter),
+        isLoadingLoraModels: isLoading,
+      }),
+    }
+  );
 
   const handleSearchFilter = useCallback((e: ChangeEvent<HTMLInputElement>) => {
     setNameFilter(e.target.value);
   }, []);
 
+  const renderModelList = (
+    filterArray: Partial<CombinedModelFormat>[],
+    isLoading: boolean,
+    loadingMessage: string,
+    title: string,
+    modelList: MainModelConfigEntity[] | LoRAModelConfigEntity[]
+  ) => {
+    if (!filterArray.includes(modelFormatFilter)) return;
+
+    if (isLoading) {
+      return <FetchingModelsLoader loadingMessage={loadingMessage} />;
+    }
+
+    if (modelList.length === 0) return;
+
+    return (
+      <StyledModelContainer>
+        <Flex sx={{ gap: 2, flexDir: 'column' }}>
+          <Text variant="subtext" fontSize="sm">
+            {title}
+          </Text>
+          {modelList.map((model) => (
+            <ModelListItem
+              key={model.id}
+              model={model}
+              isSelected={selectedModelId === model.id}
+              setSelectedModelId={setSelectedModelId}
+            />
+          ))}
+        </Flex>
+      </StyledModelContainer>
+    );
+  };
+
```

Comment:
This looks suspiciously like a component - let's move this into its own component.

Also, if a component should render nothing, it needs to explicitly return `null` - `undefined` is not a valid JSX element.

Reply:
made it a component